docs/topics/development/data_management.md

```diff
@@ -35,30 +35,46 @@ The `make initialize` command, executed as part of `make initialize_docker`, per
+You can export and load data snapshots to manage data states across different environments or for backup purposes.
+The Makefile provides commands to facilitate this.
+These commands rely internally on [django-dbbackup](https://django-dbbackup.readthedocs.io/en/stable/)
 
-- **Exporting Data**:
+- **Data dump**:
 
   ```sh
-  make data_export [EXPORT_DIR=<path>]
+  make data_dump [ARGS="--name <name> --force"]
   ```
 
-  This command creates a dump of the current MySQL database. The optional `EXPORT_DIR` argument allows you to specify a custom path for the export directory.
-  The default value is a timestamp in the `backups` directory.
+  This command creates a dump of the current MySQL database. The command accepts an optional `name` argument which will determine
+  the name of the directory created in the backup directory. By default it uses a timestamp to ensure uniqueness.
 
-  The data exported will be a .sql dump of the current state of the database including any data that has been added or modified.
+  You can also specify the `--force` argument to overwrite an existing backup with the same name.
 
 - **Loading Data**:
 
   ```sh
-  make data_restore [RESTORE_DIR=<path>]
+  make data_load [ARGS="--name <name>"]
   ```
 
-  This command restores a MySQL database from a previously exported snapshot. The optional `RESTORE_DIR` argument allows you to specify the path of the import file.
-  This must be an absolute path. It defaults to the latest stored snapshot in the `backups` directory.
+  This command will load data from an existing backup directory, synchronize the storage directory and reindex elasticsearch.
+  The name is required and must match a directory in the backup directory.
 
-Refer to the Makefile for detailed instructions on these commands.
+## Hard Reset Database
 
-This comprehensive setup ensures that the development environment is fully prepared with the necessary data.
+The actual mysql database is created and managed by the `mysqld` container. The database is created on container start
+and the actual data is stored in a persistent data volume. This enables data to persist across container restarts.
 
-By following these practices, developers can manage data effectively in the **addons-server** project. The use of persistent volumes, external mounts, data snapshots, and automated data population ensures a robust and flexible data management strategy. For more detailed instructions, refer to the project's Makefile and Docker Compose configuration in the repository.
+`addons-server` assumes that a database named `olympia` already exists and most data management commands will fail
+if it does not.
+
+If you need to hard reset the database (for example, to start with a fresh state), you can use the following command:
+
+```bash
+make down && docker_mysqld_volume_remove
```

Comment:
This fails with "docker_mysql_volume_remove: command not found".  If you run the commands separately the second fails because `make down` stops the `web` container that `docker_mysqld_volume_remove` relies on.

Reply:
Does that actually work for you? `make down` stops the `web` container that `make docker_mysqld_volume_remove` relies on

Reply:
you do not need the container running to remove the volume, yes it works. To be clean I'm talking about the updated code

```sh
make down && make docker_mysqld_volume_remove
```

Reply:
Okay, it's working now (not sure why it didn't earlier).  

It exposes another bug though - `make docker_mysqld_volume_remove` deletes the database, but we don't now call `create_db` anywhere so there isn't a database.

Reply:
The DB is created by the mysql container on startup. You only need to create the volume which happens during make up automatically. Initialize then handles seeding and loading data based on the state of the DB and your arguments.

Reply:
It failed the first time because of a missing `make` in the second command.

Reply:
```
% make down && make docker_mysqld_volume_remove
docker compose down --rmi local --remove-orphans --volumes
[+] Running 17/17
 ✔ Container olympia-worker-1           Removed                                                                                                                                         10.4s 
 ✔ Container olympia-rabbitmq-1         Removed                                                                                                                                          6.9s 
 ✔ Container olympia-elasticsearch-1    Removed                                                                                                                                          0.7s 
 ✔ Container olympia-nginx-1            Removed                                                                                                                                          0.5s 
 ✔ Container olympia-memcached-1        Removed                                                                                                                                          0.5s 
 ✔ Container olympia-mysqld-1           Removed                                                                                                                                          4.0s 
 ✔ Container olympia-redis-1            Removed                                                                                                                                          0.5s 
 ✔ Container olympia-autograph-1        Removed                                                                                                                                          0.6s 
 ✔ Container olympia-web-1              Removed                                                                                                                                         10.4s 
 ✔ Container olympia-addons-frontend-1  Removed                                                                                                                                          0.4s 
 ✔ Volume olympia_storage               Removed                                                                                                                                          0.0s 
 ✔ Image olympia-customs:latest         Removed                                                                                                                                          0.0s 
 ✔ Volume olympia_data_rabbitmq         Removed                                                                                                                                          0.0s 
 ✔ Volume olympia_data_olympia          Removed                                                                                                                                          0.0s 
 ✔ Volume olympia_data_elasticsearch    Removed                                                                                                                                          0.0s 
 ✔ Volume olympia_data_redis            Removed                                                                                                                                          0.0s 
 ✔ Network olympia_default              Removed                                                                                                                                          0.2s 
docker image prune --filter "dangling=true" --force
Total reclaimed space: 0B
docker volume prune --force
Total reclaimed space: 0B
docker volume rm addons-server_data_mysqld
Error response from daemon: remove addons-server_data_mysqld: volume is in use - [ceb8902b2d39250b76abe6a411167e6c037acee37b83a25a8a05eb3c972a9411]
make: *** [Makefile-os:115: docker_mysqld_volume_remove] Error 1
```

Reply:
@diox had you already run make down? the volume is used by mysql so if the container is running docker will complain. so long as mysql is not running you should be able to remove the volume.

Reply:
Can you try running

```sh
docker ps -a --filter volume=addons-server_data_mysqld --format "{{.ID}}: {{.Names}}"
```

This should tell us which container is using it.